internal/app/user.go

```diff
@@ -0,0 +1,124 @@
+package app
+
+import (
+	"database/sql"
+	"encoding/json"
+	ihttp "github.com/TheVitik/sport-together-go/internal/http"
+	"github.com/dgrijalva/jwt-go"
+	"net/http"
+	"time"
+)
+
+// secretKey is the secret key used to sign JWT tokens
+var secretKey = []byte("secret")
+
+// Login authenticates a user and returns a JWT token
+func Login(db *sql.DB) http.HandlerFunc {
```

Comment:
It is a bad practice to inject common, frequently used dependencies in this way. It is better to do it this way:
```
type Handler struct {
    db *sql.DB
}
func (h *Handler) Login (w http.ResponseWriter, r *http.Request) {...}
```

Reply:
Also in this way you can avoid this weird logic with `return handler `